### DISCUSSION TITLE
Xstate/inspect Vue Stuck on Waiting for Connection

### DISCUSSION THREAD

**Discussion by ghost** [statelyai/xstate#2015] [Q&A]
**Description**
I'm trying to add xstate/inspect into a vue/webpack app. But everytime I try it, nothing is happening in the xstate/inspect window. It just displays 'Waiting for Connection'. I'll include snippets where I'm putting the xstate/inspect relevant code. Trying to create a codepen to replicate this but I'm not sure familiar with it

I've put this into our main.js file 
Main.js
```
import App from './App.vue'
import Vue from 'vue'
import { vuetify, i18n } from 'common/config/vuetify'
import VueLogger from 'vuejs-logger'
import VueRouter from 'vue-router'
import 'vue-toastification/dist/index.css'
import { inspect } from '@xstate/inspect'


inspect({
  iframe: false
})

```
added this to one of the main machines.
AuthMachine.js
`export default interpret(authMachine, { devTools: true }).start()`

That's all the code I've added to the repo. It didn't seem like I needed to add anything else (unless I'm missing somthing). Is there something different you need to add if it's a vue repo using webpack instead of vue-cli?

Thanks

**Comment by davidkpiano** [comment]
Try running `inspect(...)` _before_ you `import App`.

**Reply by ghost** [reply]
hmm, no change. I'll try to just pair this down to a really simple app to see if there's any change

Edit: I think I've replicated this issue using the simple toggleMachine example in the docs. It seems if you switch from using the `useMachine(toggleMachine, { devTools: true})` to exporting the machine with `export default interpret(toggleMachine, {devTools: true}).start()` it gets the waiting for connection error.

Also switched to using toggleMachine.subscribe() instead of state.matches and vue 3 composition api style (since that is what our app currently does)

**Reply by ghost** [reply]
This code sandbox shows the issue
https://codesandbox.io/s/musing-shamir-phoqz?file=/src/main.js

**Reply by ghost** [reply]
Confirmed on our app that switching to useMachine() and vue composition API fixed it. Is the .subscribe and interpret method not supported by xstate inspect?

Trying it by using `export default interpret(authMachine, {devTools: true}).start()` and the `useService(AuthService)` method did not work and continued to see Waiting for connection error

**Reply by ghost** [reply]
Looked through the initial announcement page for xstate/inspect and found this earlier response from you! Looks like it fixed the problem! Didn't realize the .start() needed to happen in the useService 

"This one is subtle, but the rule is that inspect(...) must be called before any service is started. So if you change your code to this:

export const toggleMachineService = interpret(toggleMachine, {
  devTools: true,
-}).start();
+});

export const useToggleMachine = () => {
- return useService(toggleMachineService);
+ return useService(toggleMachineService.start());
};
Then it will work."
